salt/modules/boto_dynamodb.py

```diff
@@ -79,12 +85,132 @@ def __virtual__():
     """
     Only load if boto libraries exist.
     """
-    has_boto_reqs = salt.utils.versions.check_boto_reqs(check_boto3=False)
+    has_boto_reqs = salt.utils.versions.check_boto_reqs()
     if has_boto_reqs is True:
         __utils__["boto.assign_funcs"](__name__, "dynamodb2", pack=__salt__)
     return has_boto_reqs
 
 
+def list_tags_of_resource(
+    resource_arn, region=None, key=None, keyid=None, profile=None
+):
+    """
+    Returns a dictionary of all tags currently attached to a given resource.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_dynamodb.list_tags_of_resource \
+              resource_arn=arn:aws:dynamodb:us-east-1:012345678901:table/my-table
+    """
+    conn3 = __utils__["boto3.get_connection"](
+        "dynamodb", region=region, key=key, keyid=keyid, profile=profile
+    )
+    retries = 10
+    sleep = 6
+    tags = []
+    while retries:
+        try:
+            log.debug("Garnering tags of resource %s", resource_arn)
+            marker = ""
+            while marker is not None:
+                ret = conn3.list_tags_of_resource(
+                    ResourceArn=resource_arn, NextToken=marker
+                )
+                tags += ret.get("Tags", [])
+                marker = ret.get("NextToken")
+            return {tag["Key"]: tag["Value"] for tag in tags}
+        except botocore.exceptions.ParamValidationError as err:
+            raise SaltInvocationError(str(err))
+        except botocore.exceptions.ClientError as err:
+            if retries and err.response.get("Error", {}).get("Code") == "Throttling":
+                retries -= 1
+                log.debug("Throttled by AWS API, retrying in %s seconds...", sleep)
+                time.sleep(sleep)
+                continue
+            log.error(
+                "Failed to list tags for resource %s: %s", resource_arn, err.message
+            )
+            return None
```

Comment:
I think it makes more sense to return False here to keep in line with the other ffunctions.